profiles/hooks/addCurrentProfile.js

```diff
@@ -0,0 +1,17 @@
+const merge = require('ramda/src/merge')
+const isNil = require('ramda/src/isNil')
+const path = require('ramda/src/path')
+
+module.exports = function addCurrentProfile (hook) {
+  if (isNil(path(['params', 'credential', 'agentId'], hook))) return hook
```

Comment:
as a ramda tip, you can do this at the top-level scope (outside the function):

```js
const hasAgentId = pipe(
  path(['params', 'credential', 'agentId']),
  isNil
)
```

then call `hasAgentId(hook)` inside the hook function.